ui/app/src/components/shared/create-update-resource/CreateUpdateResource.tsx

```diff
@@ -63,18 +63,26 @@ export const CreateUpdateResource: React.FunctionComponent<CreateUpdateResourceP
     creating: ''
   }
 
-  // Construct API path for templates of specified resourceType
-  let templatesPath;
+  // Construct API paths for templates of specified resourceType
+  let templateListPath;
+  let templateGetPath;
+
+  let workspaceApplicationIdURI = undefined
   switch (props.resourceType) {
     case ResourceType.Workspace:
-      templatesPath = ApiEndpoint.WorkspaceTemplates; break;
+      templateListPath = ApiEndpoint.WorkspaceTemplates; templateGetPath = templateListPath; break;
     case ResourceType.WorkspaceService:
-      templatesPath = ApiEndpoint.WorkspaceServiceTemplates; break;
+      templateListPath = ApiEndpoint.WorkspaceServiceTemplates; templateGetPath = templateListPath; break;
     case ResourceType.SharedService:
-      templatesPath = ApiEndpoint.SharedServiceTemplates; break;
+      templateListPath = ApiEndpoint.SharedServiceTemplates; templateGetPath = templateListPath; break;
     case ResourceType.UserResource:
       if (props.parentResource) {
-        templatesPath = `${ApiEndpoint.WorkspaceServiceTemplates}/${props.parentResource.templateName}/${ApiEndpoint.UserResourceTemplates}`; break;
+        // If we are creating a user resource, parent resource must have a workspaceId
+        const workspaceId = (props.parentResource as WorkspaceService).workspaceId
```

Comment:
@damoodamoo  I tried to use workspaceCtx as we discussed but for some reason it was empty when I did my tests, so I did it the other way

Reply:
yes of course, apologies that was a bad tip - the Create form is 'global', so is above and outside of the workspace context, good call.